src/internet_identity/src/activity_stats/event_stats/tests.rs

```diff
@@ -0,0 +1,364 @@
+            session_duration_ns: Duration::from_secs(sess_duration_secs).as_nanos() as u64,
+        }),
+    };
+
+    update_events_internal(event.clone(), TIMESTAMP, &mut storage);
+    update_events_internal(event.clone(), TIMESTAMP + DAY_NS * 30, &mut storage);
+
+    assert_eq!(storage.event_data.len(), 1);
+    assert_eq!(storage.event_aggregations.len(), 4);
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_count_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1
+    );
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        sess_duration_secs
+    );
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_count_30d_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1
+    );
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_30d_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        sess_duration_secs
+    );
+}
+
+#[test]
+fn should_account_for_dapps_changing_session_lifetime() {
+    let mut storage = test_storage();
+    let event1 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(1800).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event1, TIMESTAMP, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1800
+    );
+
+    let event2 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(900).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event2, TIMESTAMP + 1, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        2700
+    );
+
+    let event3 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(1).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event3, TIMESTAMP + 1 + DAY_NS, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1
+    );
+}
+
+#[test]
+fn should_remove_zero_weighted_aggregations() {
```

Comment:
Sorry, but I don't understand what is being tested here. Why are the aggregations for example.com removed?

Is it because the next events added are older than 30 days?